src/ml_tooling/baseclass.py

```diff
@@ -327,7 +233,12 @@ def save_estimator(
         return estimator_file
 
     def make_prediction(
-        self, input_data: Any, proba: bool = False, use_index: bool = False
+        self,
+        data: DataSet,
+        *args,
+        proba: bool = False,
+        use_index: bool = False,
+        **kwargs,
     ) -> pd.DataFrame:
         """Makes a prediction given an input. For example a customer number.
         Passed to the implemented :meth:`get_prediction_data` method and calls `predict()`
```

Comment:
Not quite accurate. I guess it should be something with the the `DataSet` exposing `get_prediction_data`